django/cantusdb_project/main_app/views/views.py

```diff
@@ -746,28 +747,50 @@ def json_node_export(request, id: int) -> JsonResponse:
 
 def notation_json_export(request, id: int) -> JsonResponse:
     """
-    Return all fields of the notation with the specified ID
+    Return JsonResponse containing several key:value pairs
+    for the notation with the specified ID
     """
 
-    notation_qs: QuerySet[Notation] = Notation.objects.filter(id=id)
-    if len(notation_qs) == 0:
-        return HttpResponseNotFound()
+    notation: Notation = get_object_or_404(Notation, id=id)
 
-    values: dict = dict(*notation_qs.values())
-    return JsonResponse(values)
+    User = get_user_model()
+    created_by: Optional[User] = notation.created_by
```

Comment:
Sorry -- I can't resist. :)

You don't need to get the user model. If you want it only for the type annotation, it should be imported. But you only need the ID, which, as it happens, is actually the value of the field. So this should be all you need:

`"created_by": notation.created_by_id`

(&tc.)

Reply:
This is a useful bit of light django magic! Thank you!